src/Workspaces/Core/Portable/Workspace/Solution/Solution.CompilationTracker.cs

```diff
@@ -648,6 +652,29 @@ public CompilationInfo(Compilation compilation, bool hasSuccessfullyLoadedTransi
                 }
             }
 
+            private Compilation UpdateCompilationWithNewReferencesAndRecordAssemblySymbols(Compilation compilation, List<MetadataReference> newReferences, Dictionary<MetadataReference, ProjectId> metadataReferenceToProjectId)
+            {
+                if (!Enumerable.SequenceEqual(compilation.ExternalReferences, newReferences))
+                {
+                    compilation = compilation.WithReferences(newReferences);
+                }
+
+                // TODO: Record source assembly to project mapping
+                // RecordSourceOfAssemblySymbol(compilation.Assembly, this.ProjectState.Id);
```

Comment:
There will be another PR that will uncomment this. It relies on some other changes involving a new public ISourceAsseblySymbol type, and we didn't want to block fixing this bug on API ratification.